collect_app/src/main/java/org/odk/collect/android/widgets/RatingWidget.java

```diff
@@ -14,92 +14,68 @@
+
     Integer answer;
 
-    public RatingWidget(Context context, QuestionDetails questionDetails) {
+    public RatingWidget(Context context, QuestionDetails questionDetails, RangeQuestion rangeQuestion) {
         super(context, questionDetails);
 
-        RangeQuestion rangeQuestion = (RangeQuestion) getFormEntryPrompt().getQuestion();
-        int numberOfStars = rangeQuestion.getRangeEnd().intValue();
-        int columns = calculateColumns();
-        int rows = (int) Math.ceil((double) numberOfStars / columns);
-
-        gridLayout = new GridLayout(context);
-        gridLayout.setLayoutParams(new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
-
-        renderGrid(context, numberOfStars, columns, rows);
-
-        String answerText = questionDetails.getPrompt().getAnswerText();
-        if (answerText != null) {
-            answer = Integer.parseInt(answerText);
-            for (int i = 0; i < answer; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(R.drawable.ic_star);
-            }
-        }
-
-        addAnswerView(gridLayout, WidgetViewUtils.getStandardMargin(context));
+        numberOfStars = rangeQuestion.getRangeEnd().intValue();
     }
 
-    private void renderGrid(Context context, int numberOfStars, int columns, int rows) {
-        gridLayout.setColumnCount(columns);
-        gridLayout.setRowCount(rows);
-
-        for (int column = 0,
-             starId = 0;
-             starId < numberOfStars;
-             column++, starId++) {
-            column = column == columns ? 0 : column;
-
-            ImageButton imageButton = createImageButton(context, numberOfStars, starId);
-            gridLayout.addView(imageButton);
-        }
-    }
-
-    private ImageButton createImageButton(Context context, int numberOfStars, int total) {
-        ImageButton imageButton = new ImageButton(context);
-        imageButton.setImageResource(R.drawable.ic_star_border);
-        imageButton.setId(total);
-        imageButton.setPadding(0, 0, 0, 0);
-        imageButton.setBackground(null);
-        imageButton.setEnabled(!getFormEntryPrompt().isReadOnly());
-        imageButton.setOnClickListener(view -> {
-            int position = view.getId();
-            for (int i = 0; i < numberOfStars; i++) {
-                ImageButton button = (ImageButton) gridLayout.getChildAt(i);
-                button.setImageResource(i <= position ? R.drawable.ic_star : R.drawable.ic_star_border);
+    @Override
+    protected View onCreateAnswerView(Context context, FormEntryPrompt prompt, int answerFontSize) {
+        binding = RatingWidgetAnswerBinding.inflate(((Activity) context).getLayoutInflater());
+        View answerView = binding.getRoot();
+
+        binding.ratingBar.setLayoutParams(new LinearLayout.LayoutParams(LayoutParams.WRAP_CONTENT, LayoutParams.WRAP_CONTENT));
+        binding.ratingBar.setNumStars(numberOfStars);
```

Comment:
I wanted to display the number of stars as defined in the `RangeQuestion` but `setNumStars()` here doesn't seem to work. It fails to override the value mentioned in the XML file. I found in the docs that there are some conditions, when this works like the `layout_width` must be set to `WRAP_CONTENT`.